sympy/solvers/solvers.py

```diff
@@ -878,15 +878,11 @@ def _sympified_list(w):
             for fi in f:
                 pot = preorder_traversal(fi)
                 for p in pot:
-                    if not (p.is_number or p.is_Add or p.is_Mul) or \
-                            isinstance(p, AppliedUndef):
+                    if isinstance(p, AppliedUndef):
                         flags['dict'] = True  # better show symbols
                         symbols.add(p)
                         pot.skip()  # don't go any deeper
         symbols = list(symbols)
-        # supply dummy symbols so solve(3) behaves like solve(3, x)
```

Comment:
Since tests pass, and `solve(42)` is tested and passes I guess it is not needed. Could you please modify that test to read `assert solve(42) == solve(42, x) == []`?